src/array.jl

```diff
@@ -325,3 +332,31 @@ function getindex(A::CategoricalArray, i::Int)
 end
 
 levels!(A::CategoricalArray, newlevels::Vector) = _levels!(A, newlevels)
+
+
+function sortedmerge(A...)
+    T = Base.promote_eltype(A...)
+    m = Array{T}(0)
+    ordered = true
+
+    for a in A
```

Comment:
And here `a` could be changed to `l`. Else I keep thinking that's the actual contents of the array.